### PR TITLE
Use pkg:build_runner

### PR DESCRIPTION
**Pull request by kevmoo**
This handles updating the version and the templates

`pub run build_runner build`

Good to eat our own dogfood.

### CODE REVIEW

**Comment by kevmoo** [comment]
Also found a bug in build while I was at it: https://github.com/dart-lang/build/issues/1470

**Comment by kevmoo** [comment]
CC @jakemac53

File: build.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Read about `build.yaml` at https://pub.dartlang.org/packages/build_config
+targets:
+  $default:
+    sources:
+      include:
+        - tool/**
+        - pubspec.yaml
+        - lib/**
+        - 'templates/**'
```
Comment (chalin) [inline build.yaml]: You don't need quotes here.

**Comment by chalin** [comment]
Looking now ...

**Comment by kevmoo** [comment]
Just pushed a fix for tarvis – and quotes

File: build.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+          - lib/generators/*.dart
+        enabled: true
+
+builders:
+  stagehand:
+    # See https://github.com/dart-lang/build/issues/1423
+    import: "../../../tool/builder.dart"
+    builder_factories: ["stagehandBuilder"]
+    build_extensions: {".dart": [".g.dart"]}
+    build_to: source
```
Comment (chalin) [inline build.yaml]: Nit: In most cases, YAML syntax doesn't require quotes, and I'm quite sure you don't need quotes on lines 20-22 (well, maybe 22).

File: tool/build-check.sh [hunk 3]
```diff
@@ -9,7 +9,7 @@
 # Fast fail the script on failures.
 set -x -e -o pipefail
 
-dart tool/grind.dart build
+pub run build_runner build
```
Comment (chalin) [inline tool/build-check.sh]: 💯

File: tool/builder.dart [hunk 4]
```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
```
Comment (chalin) [inline tool/builder.dart]: Nit: 2018 :)

File: tool/src/code_generator.dart [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+import 'dart:math' as math;
+
+import 'package:build/build.dart';
+import 'package:path/path.dart' as p;
+import 'package:glob/glob.dart';
+import 'package:source_gen/source_gen.dart';
+
+const List<String> _allowedDotFiles = const <String>['.gitignore'];
+final RegExp _binaryFileTypes = new RegExp(
+    r'\.(jpe?g|png|gif|ico|svg|ttf|eot|woff|woff2)$',
```
Comment (chalin) [inline tool/src/code_generator.dart]: `ttf` --> `tiff`
`woff|woff2` --> `woff2?`

File: tool/src/code_generator.dart [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+  var lines = <String>[];
+  var index = 0;
+
+  while (index < encoded.length) {
+    var line = encoded.substring(index, math.min(index + 76, encoded.length));
+    lines.add(line);
+    index += line.length;
+  }
+
+  return lines.join('\r\n');
```
Comment (chalin) [inline tool/src/code_generator.dart]: Can't it just be `'\n'`?

File: tool/builder.dart (hunk 4 above)
Comment (kevmoo) [reply tool/builder.dart]: Not going to sweat these...

File: tool/src/code_generator.dart (hunk 5 above)
Comment (kevmoo) [reply tool/src/code_generator.dart]: This is just copy-paste. Not going to touch it here...

File: tool/src/code_generator.dart (hunk 6 above)
Comment (kevmoo) [reply tool/src/code_generator.dart]: This is just copy-paste. Not going to touch it here...

File: build.yaml (hunk 2 above)
Comment (kevmoo) [reply build.yaml]: This follows other places we have these. And our docs. Going to leave it.

**Comment by kevmoo** [comment]
We WANT to commit them. That's the point.

They are published and the user consumes them.

Having them committed makes it much easier to track diffs as things change.
Improves code reviews, etc.

**Pull request merged**